cmd/commands/wiring.go

```diff
@@ -201,3 +210,11 @@ func installKubeConfigSupport(command *cobra.Command, client *core.Client, kc *c
 		return nil
 	}
 }
+
+type buildFactory struct {
+	bf *pack.BuildFactory
+}
+
+func (bf *buildFactory) BuildConfigFromFlags(f *pack.BuildFlags) (core.Build, error) {
+	return bf.bf.BuildConfigFromFlags(f)
```

Comment:
Why the double indirection here? Can't we use (and mock, I assume) `pack.BuildFactory` directly?

Reply:
I started with that approach but couldn't get it to work. The problem is the signature of the `BuildFactory` method `BuildConfigFromFlags(f *BuildFlags) (*BuildConfig, error)`. If we mock this directly, the mock has to return a `*BuildConfig`, but we need to mock the `Run` method of `*BuildConfig`, so we need to return an interface which can then be mocked. I'm open to other solutions...